### PR TITLE
fix: Remove pydantic constraint <2.10.0 and update timedelta validator, serializer type hints

### PR DESCRIPTION
**Pull request by Pijukatel**
Seems like new 2.10.0 version made some changes that made our type hints cause errors. On the other hand, it seems our type hints were too generic. Making them more specific fixes the issue in new pydantic versions.

### CODE REVIEW

File: src/crawlee/_utils/models.py
Comment (Pijukatel) [inline src/crawlee/_utils/models.py:27]: Same as _timedelta_to_ms comment

File: src/crawlee/_utils/models.py
Comment (Pijukatel) [inline src/crawlee/_utils/models.py:12]: We could be more strict and require float, but I have a feeling, that None could be used for something connected with defaults somewhere in code???

Comment (vdusek) [reply src/crawlee/_utils/models.py:12]: I guess we can consider using only floats.

However, in `StatisticsState` and `StatisticsPersistedState`, we have fields like this:

```python
request_min_duration: Annotated[timedelta_ms | None, Field(alias='requestMinDurationMillis')] = None
```

Make sure it won't break them.

Comment (Pijukatel) [reply src/crawlee/_utils/models.py]: I tried adding some tests (but not sure whether I should keep them) and I think it works as expected. I am not really so sure why the previous implementation allowed **any** in so many places in this file, so I am little afraid there is some non obvious edge case I could not think of...

**Review by vdusek — APPROVED** [review]
LGTM, but let's wait for Honza's approval as well

File: tests/unit/_statistics/test_statistics_models.py
Comment (janbuchar) [inline tests/unit/_statistics/test_statistics_models.py:1]: I appreciate testing `timedelta_ms`, but there's no need to pull `StatisticsState` with it - I'd just define a new model in the test file and test that.

File: src/crawlee/_utils/models.py
Comment (janbuchar) [inline src/crawlee/_utils/models.py:22]: Was this the problematic type hint? :slightly_smiling_face:

**Review by janbuchar — CHANGES_REQUESTED** [review]
Just some minor things, otherwise it's great!

File: src/crawlee/_utils/models.py
Comment (Pijukatel) [reply src/crawlee/_utils/models.py:22]: I think it is still Pydantic bug, somewhere down there. I did not spent effort on trying to figure it out once I saw it is working well with more specific model.
I can reproduce it if I use new Pydantic version and for example in the new tests I created I do not import **Any** in the test file. It will then fail on importing **Any** even if I do not use it in the test files and it is needed only in model file.

Comment (Pijukatel) [reply src/crawlee/_utils/models.py:22]: It basically goes all the way to problem with the way annotations are evaluated: https://github.com/pydantic/pydantic/issues/2678

When you use **from __future__ import annotations** , it can break pydantic :D in those files

Comment (janbuchar) [reply src/crawlee/_utils/models.py:22]: That sounds scary, future annotations should be the norm 😃 can you elaborate a bit about what the problem is?

File: src/crawlee/_utils/models.py
Comment (Pijukatel) [reply src/crawlee/_utils/models.py]: Well that can also break pydantic. See new commit that allows | None, but has to use Union due to known issues that Pydantic can have with from __future__ import annotations and | symbol not being available for older Python versions in type hints...

File: pyproject.toml
Comment (janbuchar) [inline pyproject.toml:167]: I'd rather put that at the start of the file... this seems kinda fragile.

File: tests/unit/_utils/test_timedelata_ms.py
Comment (janbuchar) [inline tests/unit/_utils/test_timedelata_ms.py:1]: While you're at it, could you also test infinity and some normal value?

**Review by janbuchar — CHANGES_REQUESTED** [review]
Probably the last two nits :slightly_smiling_face:

File: src/crawlee/_utils/models.py
Comment (Pijukatel) [reply src/crawlee/_utils/models.py]: I do not think I can elaborate. This requires knowledge I do not posses. Let's wait with this then and see if Pydantic people can explain it: https://github.com/pydantic/pydantic/issues/11004

File: tests/unit/_utils/test_timedelata_ms.py
Comment (Pijukatel) [reply tests/unit/_utils/test_timedelata_ms.py:1]: Done

File: pyproject.toml
Comment (Pijukatel) [reply pyproject.toml:167]: Pydantic people solved the issue I reported. So I will just wait for thew new release and then this will not longer be needed.

File: tests/unit/_utils/test_timedelata_ms.py
Comment (Pijukatel) [inline tests/unit/_utils/test_timedelata_ms.py:27]: I considered 3600 * 24 - 1 to be more readable than 86399. Since it is just a test, I avoided defining SECONDS_IN_HOUR and HOURS_IN_DAYS ....

Comment (janbuchar) [reply tests/unit/_utils/test_timedelata_ms.py:27]: Thanks for avoiding that :slightly_smiling_face: I usually do `timedelta(days=3, hours=8).total_seconds()` instead.

File: src/crawlee/_utils/models.py
Comment (janbuchar) [reply src/crawlee/_utils/models.py:12]: Are you sure that the pipe syntax will work with Pydantic 3.9, for example?

**Pull request merged**
